### PR TITLE
[Mobile Payments] Tap to Pay Try a Payment flow

### LINKED ISSUE #9202: [Mobile Payments] Add Try a Payment prompt screen
## Description

In Milestone 2 of sTAP away, we're working on the first time user experience of Tap to Pay on iPhone.

In #9198 we added a set up complete screen, which confirms that Tap to Pay on iPhone is ready to use, and tells them about where they can go to use the reader.

The `Done` button on that screen should move to a new screen which prompts the merchant to try taking a payment with their own card.

## Design
iOS-Set up - test payment

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #9202 

## Description

This PR adds a new Try a Payment screen to the Set up Tap to Pay on iPhone flow. The new screen invites and informs the user about trying a payment out, and the `Try a Payment` button opens the Simple Payments flow with a pre-set amount of $0.50.

When the user completes their trial payment, we show the Order details for their payment, so they can do a refund.

## Testing instructions

Using an iPhone XS or newer, running iOS 16.0+, and a US-based store:

1. Navigate to `Menu > Payments > Set up Tap to Pay on iPhone`
2. Follow the flow until you get to the `Try a Payment` screen 
3. Tap `Try a Payment` 
4. Follow the simple payments flow, and tap `Card` on the Payment Methods screen (in a future PR, this will be split in to `Tap to Pay on iPhone` and `Card Reader`)
5. Pay with a test card
6. Observe that after you tap `Save receipt and continue`, or any other button on the final alert, the Order Details are shown
7. Refund the order
8. Observe that the refund completes successfully
9. Tap Done

Repeat to try the `Skip` button
1. Follow the flow until you get to the `Try a Payment` screen (N.B. if you've not left the app since the first time, you'll go straight to `Setup Complete`)
2. Tap `Skip`
3. Observe that the flow is closed and you're back on the Payments menu 

## Screenshots

https://user-images.githubusercontent.com/2472348/226418882-28f8fd55-28f2-4bd3-9e83-4ce6c3b98e0a.mp4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift [hunk 1]
```diff
@@ -47,7 +44,13 @@ final class SetUpTapToPayCompleteViewModel: PaymentSettingsFlowPresentedViewMode
     /// Notifies the viewModel owner if a change occurs via didChangeShouldShow
     ///
     private func reevaluateShouldShow() {
-        let newShouldShow: CardReaderSettingsTriState = connectedReader
+        let newShouldShow: CardReaderSettingsTriState
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift:47]: Non-blocker: This could be me, but I found this bit hard to follow with the `newShouldShow` name. How about something more explicit like `shouldShowNewConnectedReader` , `shouldShowNewViewModel`, `shouldShowNewViewModelAndView`or similar ?

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+private extension SetUpTapToPayPaymentPromptView {
+    enum Localization {
+        static let setUpTryPaymentPromptTitle = NSLocalizedString(
+            "Try a payment",
+            comment: "Settings > Set up Tap to Pay on iPhone > Try a Payment > Inform user that " +
+            "Tap to Pay on iPhone is ready"
+        )
+
+        static let setUpTryPaymentPromptDescription = NSLocalizedString(
+            "Try taking a payment of $0.50 to see how Tap to Pay on iPhone works. Use your " +
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift:144]: Non-blocker: 

```suggestion
            "Try taking a payment of USD $0.50 to see how Tap to Pay on iPhone works. Use your " +
```
While we support only USD at the moment, using `$` can cause confusion to folks that may use the same symbol for their currency, this happens pretty often for example for people that has their account in MXN and they're prompted to pay for a plan in USD.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+/// This view controller is used when no reader is connected. It assists
+/// the merchant in connecting to a reader.
+///
+final class SetUpTapToPayTryPaymentPromptViewController: UIHostingController<SetUpTapToPayPaymentPromptView>, PaymentSettingsFlowViewModelPresenter {
+
+    private var viewModel: SetUpTapToPayTryPaymentPromptViewModel
+
+    init?(viewModel: PaymentSettingsFlowPresentedViewModel) {
+        guard let viewModel = viewModel as? SetUpTapToPayTryPaymentPromptViewModel else {
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift:11]: Should we log a `DDLogError` here? If we happen to pass any other class that conforms to `PaymentSettingsFlowPresentedViewModel` it would fail silently.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+
+            case .failure(let error):
+                self.presentNoticeSubject.send(.error(Localization.errorCreatingTestPayment))
+                self.analytics.track(event: WooAnalyticsEvent.PaymentsFlow.paymentsFlowFailed(flow: .tapToPayTryAPayment,
+                                                                                              source: .tapToPayTryAPaymentPrompt))
+                DDLogError("⛔️ Error creating Tap to Pay try a payment order: \(error)")
+            }
+        }
+        ServiceLocator.stores.dispatch(action)
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewModel.swift]: We don't need to access to ServiceLocator here, right?
```suggestion
        stores.dispatch(action)
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift [hunk 5]
```diff
@@ -47,7 +44,13 @@ final class SetUpTapToPayCompleteViewModel: PaymentSettingsFlowPresentedViewMode
...
+        switch (doneWasTapped, connectedReader) {
+        case (true, _):
+            newShouldShow = .isFalse
+        case (false, _):
+            newShouldShow = connectedReader
+        }
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift]: `connectedReader` is not currently being used in the switch statement, I'm guessing we'll be using this to perform other logic moving forward?
```suggestion
        switch doneWasTapped {
        case true:
            newShouldShow = .isFalse
        case false:
            newShouldShow = connectedReader
        }
```

**Review by iamgabrielma — APPROVED** [review]
Works as expected. LGTM 🚢

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift (hunk 1 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift:47]: I know what you mean, and I found the same at first. However, I left it as is,  because it's named generically and matches the pattern of the other implementations. I think that actually makes it easier to read in the context of all the other screens it interacts with, and the same pattern in the ManageCardReader flow.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift (hunk 2 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift:144]: This will only show for US stores, even though I've not put that constraint on yet. In that context I don't think it should be too confusing?

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift (hunk 3 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift:11]: Good to call it out, so thank you, but it won't fail silently: this is a failable initialiser, so it will return nil and the call site will have to deal with that as appropriate.

In this case, it's the `PaymentSettingsFlowPresentingViewController`, [which will log the error when it can't make the child view controller.](https://github.com/woocommerce/woocommerce-ios/blob/dc5e351b12c4937be0c7118d6e52cd37b75aece7/WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/PaymentSettingsFlowPresentingViewController.swift#L46)

Adding another log here would be additional noise.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift (hunk 5 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewModel.swift]: I think that was my intention but it didn't end up being needed. Fixed in 885e69e

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewModel.swift (hunk 4 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewModel.swift]: Thanks, good catch. Fixed in c1169cb
